test/test_sorted_containers.jl

```diff
@@ -275,320 +275,337 @@ eq(::CaseInsensitive, a, b) = isequal(lowercase(a), lowercase(b))
 
 
     @testset "SortedDictBasic" begin
+    
         # a few basic tests of SortedDict to start
         m1 = SortedDict((Dict{String,String}()), Forward)
         @test typeof(m1) == SortedDict{String, String, ForwardOrdering}
-        kdarray = ["hello", "jello", "alpha", "beta", "fortune", "random",
-                   "july", "wednesday"]
-        checkcorrectness(m1.bt, false)
-        for i = 1 : div(size(kdarray,1), 2)
-            k = kdarray[i*2-1]
-            d = kdarray[i*2]
-            #println("- inserting: k = $k d = $d")
-            m1[k] = d
+
+        @testset "testset 1" begin
```

Comment:
It seems to me that if this doesn't separate out the tests based on the particular functionality being tested, it doesn't need to be in its own testset. If it does separate functionality, perhaps a more descriptive name than "testset 1" would be good? 😉

Reply:
Same comment for most of the rest of these; I won't go through and mark them all individually.

Reply:
For sure "testset 1" is not a very descriptive name! But it can help to catch (with a certain level of distance) where tests are broken. I'm open to suggestions for a more descriptive name for this testset and more broadly for other testsets which need a better naming.

I don't think that diff really help to understand globality of such a PR 

But a visual comparison of
https://github.com/scls19fr/DataStructures.jl/blob/418649bcbd8b476dd5cc22b31646e4d7b7eec887/test/test_sorted_containers.jl
https://github.com/JuliaCollections/DataStructures.jl/blob/master/test/test_sorted_containers.jl
can help to see the general approach

Maybe I shouldn't have tackle this this way... maybe a first should be to split `test_sorted_containers.jl` into several files (some files for unit tests and file(s) for defining functions such as `fulldump`, `checkcorrectness`, ... because a file with nearly 2000 lines of tests is something we should avoid.

With this work on this PR, I noticed that running tests of this file is pretty long, and so we should be able to run just a part of this (I also wonder if Julia tests have timing features and how we can speedup tests)

With Python and [nose](http://nose.readthedocs.io/) we have [nose-timer](https://github.com/mahmoudimus/nose-timer) maybe such a timing plugin should be developed (or enabled)